### PR TITLE
feat: removed legacy app tracking label

### LINKED ISSUE #12264: Drop support for `applications.argoproj.io/app-name`
# Summary

Remove support for old tracking label, `applications.argoproj.io/app-name`

# Motivation

`"applications.argoproj.io/app-name"` is found once in the argo-cd repo:

```go
	// LabelKeyLegacyApplicationName is the legacy label (v0.10 and below) and is superseded by 'app.kubernetes.io/instance'
	LabelKeyLegacyApplicationName = "applications.argoproj.io/app-name"
```

That's very old.

The constant is used in one place: `SetAppInstanceLabel`.

That function is called in only one place: `SetAppInstance`.

That function is called by `groupObjsForDiff`, which uses `argoSettings.AppLabelKey` for the `key` param, and by `GenerateManifests`, which uses `q.AppLabelKey` for the `key` param.

`q.AppLabelKey` is set in 4 places:

* `getLocalObjectsString`
   * called by `getLocalObjects`
      * called by `findandPrintDiff`, which uses `argoSettings.AppLabelKey`
      * called by `NewApplicationManifestsCommand`, which uses `argoSettings.AppLabelKey`
   * called by `NewApplicationSyncCommand`, which uses `argoSettings.AppLabelKey`
* `getRepoObjs`
   * called by `CompareAppState`, which uses `appStateManager.settingsMgr.GetAppInstanceLabelKey()`, which pulls it directly from `application.instanceLabelKey` in argocd-cm
* `GetManifests`, which uses `Server.settingsMgr.GetAppInstanceLabelKey()`, which gets the value directly from the configmap
* `GetManifestsWithFiles`, which does the same as `GetManifests`

I believe that is a full accounting of where the deprecated label was used. Basically, if that label isn't explicitly configured in argocd-cm, we're not using it in the code base.

# Proposal

Let's remove support for this label in v3.0.

### PR DESCRIPTION
**Pull request by gdsoumya**
Fixes https://github.com/argoproj/argo-cd/issues/12264

Checklist:

* [x] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [x] The title of the PR conforms to the [Toolchain Guide](https://argo-cd.readthedocs.io/en/latest/developer-guide/toolchain-guide/#title-of-the-pr)
* [x] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [x] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [ ] Optional. My organization is added to USERS.md.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md#legal)
* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [ ] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)). 
* [ ] My new feature complies with the [feature status](https://github.com/argoproj/argoproj/blob/master/community/feature-status.md) guidelines.
* [ ] I have added a brief description of why this PR is necessary and/or what this PR solves.

Please see [Contribution FAQs](https://argo-cd.readthedocs.io/en/latest/developer-guide/faq/) if you have questions about your pull-request.

### CODE REVIEW

File: util/kube/kube.go [hunk 1]
```diff
@@ -26,70 +22,6 @@ func SetAppInstanceLabel(target *unstructured.Unstructured, key, val string) err
 	}
 	labels[key] = val
 	target.SetLabels(labels)
-	if key != common.LabelKeyLegacyApplicationName {
```
Comment (ishitasequeira) [inline util/kube/kube.go:29]: Remove the legacy labeling statement from the function description as well. Also, as there is no chance of getting errors anymore, remove returning of error as part of function definition.

**Review by ishitasequeira — CHANGES_REQUESTED** [review]

File: util/kube/kube.go (hunk 1 above)
Comment (gdsoumya) [reply util/kube/kube.go:29]: Fixed

**Review by ishitasequeira — APPROVED** [review]
LGTM!!

File: util/kube/kube.go [hunk 2]
```diff
@@ -18,91 +14,23 @@ func IsValidResourceName(name string) bool {
...
-			}
-		}
-	}
-	return nil
 }
 
 // SetAppInstanceAnnotation the recommended app.kubernetes.io/instance annotation against an unstructured object
-// Uses the legacy labeling if environment variable is set
-func SetAppInstanceAnnotation(target *unstructured.Unstructured, key, val string) error {
+func SetAppInstanceAnnotation(target *unstructured.Unstructured, key, val string) {
```
Comment (gdsoumya) [inline util/kube/kube.go:27]: @ishitasequeira found another function with similar redundancy so refactored that too. Let me know if I should revert it.

**Review by ishitasequeira — APPROVED** [review]
Thanks @gdsoumya !! Changes look good!!

**Comment by ishitasequeira** [comment]
@gdsoumya , It seems the PR has conflicts. Do you want to rebase?

**Review by ishitasequeira — CHANGES_REQUESTED** [review]
@gdsoumya I think we missed getting this PR in. Do you mind resolving conflicts?

**Comment by gdsoumya** [comment]
@ishitasequeira resolved the conflicts.

**Review by ishitasequeira — APPROVED** [review]
LGTM!

**Review by pasha-codefresh — APPROVED** [review]
LGTM

**Review by wanghong230 — APPROVED** [review]

**Comment by crenshaw-dev** [comment]
@gdsoumya turns out Intuit is less ready to kiss this code goodbye than I hoped. I'm going to open a PR to revert the changes, please let me know if there are concerns! We'll try to make up for the continued tech debt in other code. :-)

**Pull request merged**